source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -396,6 +396,12 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
   Http::FilterDataStatus encodeData(Buffer::Instance& data, bool end_stream) override;
   Http::FilterTrailersStatus encodeTrailers(Http::ResponseTrailerMap& trailers) override;
 
+  void encodeComplete() override {
+    if (config_->observabilityMode()) {
```

Comment:
Why not move it here always for consistency?

Reply:
This is because ext_proc can still receive the response after its own `encodeComplete`. The good thing(why it works here) is because ob11y mode doesn't have/need response

I also almost forgot this since the related discussion was quite a while back. Thankfully I wrote one-pager at that time :)  I just tagged you in that doc

Reply:
Fair, makes sense.